common/djangoapps/third_party_auth/tests/specs/test_testshib.py

```diff
@@ -519,13 +521,20 @@ def test_register_http_failure_in_odata(self):
             expected_message = 'Unable to retrieve user details with username {username} from SAPSuccessFactors ' \
                                'for company ID {company_id} with url "{odata_api_url}".  Error message: ' \
                                '500 Server Error: Internal Server Error for url: {odata_api_url}.  System message: ' \
-                               'Not available.'.format(
+                               'Not available.  Headers: '.format(
                                    username=self.USER_USERNAME,
                                    company_id=odata_company_id,
                                    odata_api_url=mocked_odata_ai_url,
                                )
-            logging_messages = [log_msg.getMessage() for log_msg in log_capture.records]
-            self.assertTrue(expected_message in logging_messages)
+            logging_messages = str([log_msg.getMessage() for log_msg in log_capture.records]).replace('\\', '')
+            self.assertIn(odata_company_id, logging_messages)
+            self.assertIn(mocked_odata_ai_url, logging_messages)
+            self.assertIn("SAPSuccessFactors", logging_messages)
+            self.assertIn("Error message", logging_messages)
+            self.assertIn("System message", logging_messages)
+            self.assertIn("Headers", logging_messages)
+            self.assertIn("CorrelationId", logging_messages)
+            self.assertIn("RequestNo", logging_messages)
```

Comment:
Do we need all this? Seems like you are just testing the mocking code.

Reply:
I changed it because adding the headers to the expected message was a bit of a PITA and I actually only really care that certain information is being provided versus the actual text of the message.  I can remove the "CorrelationId" and "RequestNo" checks since you are correct that we're just testing the mocked response, although I want to leave them in the headers since they're important.